File: src/main/java/com/footverse/product/dto/ProductVariantResponse.java
package com.footverse.product.dto;

import java.math.BigDecimal;

import com.footverse.product.entity.ProductVariantStatus;

/**
 * A product variant returned to clients (dto-spec §9).
 *
 * @param id            the variant id
 * @param color         the color
 * @param size          the size
 * @param price         the effective price ({@code priceOverride}, or the product's
 *                      {@code basePrice} when {@code priceOverride} is null)
 * @param stockQuantity the available stock
 * @param status        the variant status
 * @param sku           the stock-keeping unit
 */
public record ProductVariantResponse(
        Long id,
        String color,
        String size,
        BigDecimal price,
        int stockQuantity,
        ProductVariantStatus status,
        String sku) {
}

File: admin/src/features/product/models/product-variant-response.ts
import type { ProductVariantStatus } from './product-variant-status';

/**
 * A product variant returned by the public/write endpoints (dto-spec §9) —
 * `costPrice` is ADMIN-only and is omitted here (see
 * `AdminProductVariantResponse`). Mirrors the frozen DTO field-for-field.
 * Returned by `POST/PUT /products/{id}/variants`.
 */
export interface ProductVariantResponse {
  readonly id: number;
  readonly color: string;
  readonly size: string;
  readonly price: number;
  readonly stockQuantity: number;
  readonly status: ProductVariantStatus;
  readonly sku: string;
}
